Source/WebKit/Scripts/generate-serializers.py

```diff
@@ -471,30 +475,33 @@ def parse_serialized_types(file, file_name):
                 members.append(MemberVariable(member_type, member_name, member_condition, []))
     if len(headers) == 0 and len(serialized_types) <= 1:
         headers = [ConditionalHeader('"' + file_name[0:len(file_name) - len('.serialization.in')] + '.h"', None)]
-    return [serialized_types, serialized_enums, headers]
+    return [serialized_types, serialized_enums, headers, ext]
 
 
 def main(argv):
-    serialized_types = []
-    serialized_enums = []
-    headers = []
+    serialized_types = {}
+    serialized_enums = {}
+    headers = {}
+    exts = set()
     for i in range(2, len(argv)):
         with open(argv[1] + argv[i]) as file:
-            new_types, new_enums, new_headers = parse_serialized_types(file, argv[i])
+            new_types, new_enums, new_headers, ext = parse_serialized_types(file, argv[i])
+            exts.add(ext)
             for type in new_types:
-                serialized_types.append(type)
+                serialized_types.setdefault(ext, []).append(type)
             for enum in new_enums:
-                serialized_enums.append(enum)
+                serialized_enums.setdefault(ext, []).append(enum)
             for header in new_headers:
-                headers.append(header)
-    headers.sort()
+                headers.setdefault(ext, []).append(header)
+    [v.sort() for v in headers.values()]
 
     with open('GeneratedSerializers.h', "w+") as header_output:
-        header_output.write(generate_header(serialized_types, serialized_enums))
-    with open('GeneratedSerializers.cpp', "w+") as cpp_output:
-        cpp_output.write(generate_cpp(serialized_types, serialized_enums, headers))
+        header_output.write(generate_header(sum(serialized_types.values(), []), sum(serialized_enums.values(), [])))
+    for ext in exts:
```

Comment:
Could we use "extension" or "file_extension" here?  We typically don't like unnecessary abbreviations in our variable names.  Same with "exts" above.